ext/aws/FreeRTOSConfig.h.in

```diff
@@ -36,12 +36,12 @@ your application. */
 #include <modm/platform/device.hpp>
 
 // declared in modm:platform:clock
-modm_extern_c uint32_t _ZN4modm8platform4fcpuE;
+modm_extern_c uint32_t SystemCoreClock;
```

Comment:
Should `modm_extern_c` really be empty for C? I feel like it should be `extern "C"` for C++ and `extern` for C.

Reply:
Maybe we should completely remove the `modm_extern_c` macro to prevent such confusion in future?

Reply:
Yeah, I'm not impressed by it either…

Reply:
It's used in conflicting ways as a function attribute and a variable attribute:

```
modm/board/board.cpp.in
29:modm_extern_c void

modm/architecture/interface/assert.hpp.in
172:modm_extern_c void

modm/architecture/interface/interrupt.hpp
98:		modm_extern_c void vector ## _vect(void)
109:		modm_extern_c void vector ## _IRQHandler(void)
113:		modm_extern_c void vector ## _IRQHandler(void) \
122:		modm_extern_c void vector ## _isr(void)
126:		modm_extern_c void vector ## _isr(void) __VA_ARGS__; \

modm/architecture/interface/assert.h.in
109:modm_extern_c void

modm/architecture/utils.hpp
152:	#	define modm_extern_c extern "C"
154:	#	define modm_extern_c extern
170:		modm_extern_c uint8_t name ## _laddr[];		\
171:		modm_extern_c const size_t name ## _length = (size_t) name ## _laddr; \
172:		modm_extern_c uint8_t name[]
```

Reply:
| type | C | C++ |
|---|:---|:---|
| function | `void fn()` | `extern "C" void fn()` |
| variable | `extern int var;` | `extern "C" int var` |

Help?

Reply:
So I have to have `modm_extern_c_function` and `modm_extern_c_variable` macros? WTF is this…

Reply:
Oh, yes this was only the first part of the fix. The `extern` problem remains.

Reply:
@salkinium You could try to unify the macros. Putting `extern` on a function declaration does nothing in most cases, external linkage is the default.